src/h5web/vis-packs/core/hooks.ts

```diff
@@ -143,3 +105,42 @@ export function useWheelCapture() {
 
   return ref;
 }
+
+export const useCombinedDomain = createMemo(getCombinedDomain);
+
+export function useMappedArray<T extends unknown[] | undefined>(
+  value: T,
+  dims: number[],
+  mapping: DimensionMapping,
+  autoScale?: boolean
+): T extends (infer U)[] ? [ndarray<U>, ndarray<U>] : [undefined, undefined];
+
+export function useMappedArray<T>(
+  value: T[] | undefined,
+  dims: number[],
+  mapping: DimensionMapping,
+  autoScale?: boolean
+) {
+  const baseArray = createMemo(getBaseArray)(value, dims);
+  const mappedArray = createMemo(applyMapping)(baseArray, mapping);
```

Comment:
Ooh you're recreating a hook on every call here 😅 -- just move them out above `useCombinedDomain`